fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/qrcode/ScanningAction.java

```diff
@@ -1,30 +1,110 @@
 package com.github.dedis.popstellar.ui.qrcode;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
+
+import com.github.dedis.popstellar.R;
 
 /** Enum class modeling the the action we want to do when using the QR code fragment */
 public enum ScanningAction {
```

Comment:
> I had never seen this enum 😅
> 
> As every function returns constant values, wouldn't it be easier to set them as stored fields ?
> 
> ```
> ADD_WITNESS("Add witness", R.string.qrcode_scanning_add_witness, R.string.scanned_witness, R.string.add_witness_title, R.string.manual_witness_hint);
> ```
> 
> And then have a constructor
> 
> ```
> ScanningAction(name, instructions, scanTitle, pageTitle, hint) {
>  [...]
> }
> ```
> 
> And use this fields directly instead of functions.

I personally find it easier to understand that way and less bulky. But I can change it if you have strong feelings about it. Maybe we can get @jbsv input on this ?



> Also, I don't really like the point to override toString().
> 
> If we want to show it to the user, it needs to be a string resource

Yeah that's legacy that I should have removed. It's done now :)